evidence/pool_test.go

```diff
@@ -87,21 +93,54 @@ func TestEvidencePool(t *testing.T) {
 
 func TestEvidencePoolIsCommitted(t *testing.T) {
 	// Initialization:
-	valAddr := []byte("validator_address")
-	height := int64(42)
-	stateDB := initializeValidatorState(valAddr, height)
-	evidenceDB := dbm.NewMemDB()
-	pool := NewPool(stateDB, evidenceDB)
+	var (
+		valAddr       = []byte("validator_address")
+		height        = int64(42)
+		lastBlockTime = time.Now()
+		stateDB       = initializeValidatorState(valAddr, height)
+		evidenceDB    = dbm.NewMemDB()
+		pool          = NewPool(stateDB, evidenceDB)
+	)
 
 	// evidence not seen yet:
-	evidence := types.NewMockGoodEvidence(height, 0, valAddr)
+	evidence := types.NewMockEvidence(height, time.Now(), 0, valAddr)
 	assert.False(t, pool.IsCommitted(evidence))
 
 	// evidence seen but not yet committed:
 	assert.NoError(t, pool.AddEvidence(evidence))
 	assert.False(t, pool.IsCommitted(evidence))
 
 	// evidence seen and committed:
-	pool.MarkEvidenceAsCommitted(height, []types.Evidence{evidence})
+	pool.MarkEvidenceAsCommitted(height, lastBlockTime, []types.Evidence{evidence})
 	assert.True(t, pool.IsCommitted(evidence))
 }
+
+func TestAddEvidence(t *testing.T) {
+
+	var (
+		valAddr      = []byte("val1")
+		height       = int64(100002)
+		stateDB      = initializeValidatorState(valAddr, height)
+		evidenceDB   = dbm.NewMemDB()
+		pool         = NewPool(stateDB, evidenceDB)
+		evidenceTime = time.Date(2019, 1, 1, 0, 0, 0, 0, time.UTC)
+	)
+
+	testCases := []struct {
+		height int64
```

Comment:
Could you also add a `description string` (e.g. "outdated evidence", "too much blocks have passed")